react/features/calendar-sync/middleware.js

```diff
@@ -192,7 +192,19 @@ function _getURLFromEvent(event, knownDomains) {
             const matches = urlRegExp.exec(field) || schemeRegExp.exec(field);
 
             if (matches) {
-                return matches[0];
+                const parsedURI = parseURIString(matches[0]);
+
+                return `${
+                    parsedURI.protocol
+                }//${
+                    parsedURI.host
+                }/${
+                    parsedURI.room
+                }${
+                    parsedURI.search
+                }${
+                    parsedURI.hash
+                }`;
```

Comment:
I'm very concerned about the reconstruction of the URL:

1. It's not correct because we support tenants which are part of the path.

2. I still don't understand where the app broke. Could you please explain which character got reported to you and where exactly the app broke?

3. The reconstruction seems to me like the implementation of `toString` on the object returned by `parseURIString `. The object actually has a `toString` implemented on it but, unfortunately, the object doesn't properly modify itself when you set its properties. For example, you went and modified the `room` property. The value of the `room` property though is a substring of the `pathname` property. But `room` doesn't have a setter to reflect the change into `pathname`. So I'd rather see something more substantial in this PR but only after we understand where exactly the app breaks. Let's talk after you've got more details.

Reply:
1. What are tenants in terms of jitsi? I only found a single reference of them in the code but that doesn't tell me too much.
2. The easiest to reproduce the issue is to connect to a room on meet.jit.si thats name contains some special characters, such as a comma or @. Beta is not affected (on web) because it has a more strict proxy pattern. Mobile is affected on beta too though. The root cause is that jxmpp has a list of characters that is doesn't accept in the local part: https://github.com/igniterealtime/jxmpp/blob/40397e1cf011420368a45aec805fbccec528785e/jxmpp-core/src/main/java/org/jxmpp/stringprep/simple/SimpleXmppStringprep.java#L55
So at the end the exclusion logic should be a combination of the proxy setting (Aaron agreed to use the RFC suggestion) and this list in the jxmpp lib.
3. Ok, let's talk on out catch up!

Reply:
Another suggestion, if we want to make the solution even more standard is to skip the regexp and apply encodeURIComponent on the room name. (?)